fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/MetadataController.java

```diff
@@ -0,0 +1,387 @@
+				Optional<ActuatorProvider> actuatorProvider, Optional<List<OperationCustomizer>> operationCustomizers,
+				Optional<List<OpenApiCustomiser>> openApiCustomisers,
+				SpringDocConfigProperties springDocConfigProperties,
+				Optional<SecurityOAuth2Provider> springSecurityOAuth2Provider,
+				Optional<RouterFunctionProvider> routerFunctionProvider,
+				Optional<RepositoryRestResourceProvider> repositoryRestResourceProvider) {
+			super(openAPIBuilderObjectFactory, requestBuilder, responseBuilder, operationParser, requestMappingHandlerMapping,
+					actuatorProvider, operationCustomizers, openApiCustomisers, springDocConfigProperties,
+					springSecurityOAuth2Provider, routerFunctionProvider, repositoryRestResourceProvider);
+		}
+		
+		@Override
+		public synchronized OpenAPI getOpenApi() {
+			return super.getOpenApi();
+		}
+	}
+	
+	
+	@Autowired
+	private ObjectMapper objectMapper;
+	
+	
+	@Autowired
+	private SnowOwlOpenApiWebMvcResource openApiResource;
+	
+	//RestController is singleton scoped, only a single instances of these objects are 'cached'
+	private CapabilityStatement capabilityStatement;
+	private Map<String, OperationDefinition> operationMap = Maps.newHashMap();
+
+	/**
+	 * Returns the server's capability statement
+	 * @param request the injected http servlet request
+	 * @return
+	 */
+	@Operation(
+			summary="Retrieve the capability statement", 
+			description="Retrieves this server's capability statement.",
+			extensions = {
+				@Extension(name = B2I_OPENAPI_X_INTERACTION, properties = {
+					@ExtensionProperty(name = B2I_OPENAPI_INTERACTION_READ, value = "Read the capability statement"),
+				}),
+			}
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+	})
+	@GetMapping(value="/metadata")
+	public Promise<CapabilityStatement> metadata(
+			//@io.swagger.v3.oas.annotations.Parameter(extensions = @Extension(name = "dsd", 
+			//properties = @ExtensionProperty(name = "type", value = "String")), 
+			//description = "Canonical URL of the value set")
+			
+			HttpServletRequest request) {
+		
+		if (capabilityStatement == null) {
+			initCache(request);
+		}
+		return Promise.immediate(capabilityStatement);
+	}
+	
+	/**
+	 * Returns the {@link OperationDefinition} for a given operation.
+	 * @param operation
+	 * @return
+	 */
+	@Operation(
+			summary="Retrieve an operation definition by its name", 
+			description="Retrieves an operation definition by its compound name (resource$operation)."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "404", description = "Operation definition not found")
+	})
+	@GetMapping(value="/OperationDefinition/{operation}")
+	public Promise<OperationDefinition> operationDefinition(
+			@PathVariable(value = "operation") final String operation,
+			HttpServletRequest request) {
+		
+		if (capabilityStatement == null) {
+			initCache(request);
```

Comment:
This method call requires double-checked locking otherwise two or more threads might reinitialize both fields unnecessarily.